src/balancereader/src/test/java/anthos/samples/bankofanthos/balancereader/BalanceReaderControllerTest.java

```diff
@@ -0,0 +1,193 @@
+    private LedgerReader ledgerReader;
+    @Mock
+    private DecodedJWT jwt;
+    @Mock
+    private Claim claim;
+    @Mock
+    private TransactionRepository transactionRepository;
+    @Mock
+    private LoadingCache<String, Long> cache;
+
+    private static final String VERSION = "v0.2.0";
+    private static final String LOCAL_ROUTING_NUM = "123456789";
+    private static final int CACHE_SIZE = 100000;
+    private static final String OK_CODE = "ok";
+    private static final String JWT_ACCOUNT_KEY = "acct";
+    private static final long BALANCE = 100l;
+    private static final String AUTHED_ACCOUNT_NUM = "1234567890";
+    private static final String NON_AUTHED_ACCOUNT_NUM = "9876543210";
+    private static final String BEARER_TOKEN = "Bearer abc";
+    private static final String TOKEN = "abc";
+
+    @BeforeEach
+    void setUp() {
+        initMocks(this);
+        balanceReaderController = new BalanceReaderController(ledgerReader, verifier, cache, LOCAL_ROUTING_NUM, VERSION);
+
+        when(verifier.verify(TOKEN)).thenReturn(jwt);
+        when(jwt.getClaim(JWT_ACCOUNT_KEY)).thenReturn(claim);
+    }
+
+    @Test
+    @DisplayName("Given version number in the environment, " +
+            "return a ResponseEntity with the version number")
+    void version() {
+        // When
+        final ResponseEntity actualResult = balanceReaderController.version();
+
+        // Then
+        assertNotNull(actualResult);
+        assertEquals(VERSION, actualResult.getBody());
+        assertEquals(HttpStatus.OK, actualResult.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Given the server is serving requests, return HTTP Status 200")
+    void readiness() {
+        // When
+        final String actualResult = balanceReaderController.readiness();
+
+        // Then
+        assertNotNull(actualResult);
+        assertEquals(OK_CODE, actualResult);
+    }
+
+    @Test
+    @DisplayName("Given the ledgerReader is alive, return HTTP Status 200")
+    void livenessSucceedsWhenLedgerReaderIsAlive() {
+        // Given
+        when(ledgerReader.isAlive()).thenReturn(true);
+
+        // When
+        final ResponseEntity actualResult = balanceReaderController.liveness();
+
+        // Then
+        assertNotNull(actualResult);
+        assertEquals(OK_CODE, actualResult.getBody());
+        assertEquals(HttpStatus.OK, actualResult.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Given the ledgerReader is not alive, return HTTP Status 500")
+    void livenessFailsWhenLedgerReaderIsNotAlive() {
+        // Given
+        when(ledgerReader.isAlive()).thenReturn(false);
+        
+        // When
+        final ResponseEntity actualResult = balanceReaderController.liveness();
+
+        // Then
+        assertNotNull(actualResult);
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, actualResult.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Given the user is authenticated for the account, return HTTP Status 200")
```

Comment:
update title to mention balance as well: "returns HTTP Status 200 and expected balance".  It might make sense to separate these into two tests too, WDYT?

Reply:
True, if somehow the endpoint didn't fail but returned the wrong balance, this test wouldn't be very helpful. I'll split them into separate tests